File: server/src/main/java/lighting/server/scene/SceneService.java
package lighting.server.scene;

import java.io.IOException;

public interface SceneService {

	boolean recordScene(int buttonId);

	boolean playSceneFromButton(int buttonId) throws IOException;

	boolean playSceneFromId(String sceneId) throws IOException;

	boolean recordSceneMultipleFrames(int buttonId);

	boolean stopRecording();

	void stop();

	void pause(boolean bool);
}

File: server/src/main/scala/lassi/server/scene/SceneService.scala
package lassi.server.scene

trait SceneService {

  def recordScene(buttonId: Int): Boolean

  def playSceneFromButton(buttonId: Int): Boolean

  def playSceneFromId(sceneId: String): Boolean

  def recordSceneMultipleFrames(buttonId: Int): Boolean

  def stopRecording: Boolean

  def stop(): Unit

  def pause(bool: Boolean): Unit
}
